### ISSUE TITLE
Some Questions about weird behavior(s)

### ISSUE THREAD

**Issue by rkdrns4747** [Tenpi/waifu2x#17]
1. I got this kind of weird rainbow line on the stair-like lines, and is there any way to handle this?

- source
![1](https://user-images.githubusercontent.com/45228785/160289265-1794b2ba-c5fa-4ad5-915c-a04bdbb073be.png)

- 5.12x upscaled

![1-5 12x2](https://user-images.githubusercontent.com/45228785/160289217-f9d15f39-71c1-4cf5-9bca-404d23e052ca.png)

2. GIF upscaling got really weird behavior (some frame could not handle transparency, and some frame lost lot's of pixels):

2-1 -> some frame could not handle transparency

- source
![1](https://user-images.githubusercontent.com/45228785/160289527-5194e284-ca9c-4b43-9eb9-66742fde6ae3.gif)

- 5.12 upscaled 
![1-5 12x2](https://user-images.githubusercontent.com/45228785/160289546-a073ce90-1e21-4640-ae3c-a570c11b94dd.gif)

2-2-> some frame lost lot's of pixels
- source
![5](https://user-images.githubusercontent.com/45228785/160289418-df431f67-4cae-46f5-ad48-07bab1e37653.gif)

- 5.12x upscaled
![5-5 12x2](https://user-images.githubusercontent.com/45228785/160289474-d74f095f-d400-415c-9b04-04b876674a1b.gif)

**Comment by Moebytes** [comment]
a) This is an artifact of the waifu2x algorithm, nothing I can do about it.

b) I have updated it to replace `transparency` with `transparentColor`, so if you set it to `#000000` for example then black will be the transparent color, avoiding that problem. 

Overall I have found that the GIFEncoder does a poor job when encoding transparent gifs, I will see if I can find a better gif encoder.

c) This is caused by setting `cumulative` to `false` in the gif options (or not specifying it). Some gifs only store partial frame data per frame, so setting cumulative to `true` fixes this issue by layering it with previous frames. I have made it true by default in the latest version. 

I recommend converting your transparent gifs to animated webp because the webp encoder is much better. The function is `waifu2x.upscaleAnimatedWebp()`, with same parameters as `upscaleGIF()` function.
